packages/orchestration/src/exos/chain-hub.js

```diff
@@ -242,12 +243,15 @@ const ChainHubI = M.interface('ChainHub', {
  * @param {Zone} zone
  * @param {Remote<NameHub>} agoricNames
  * @param {VowTools} vowTools
+ * @param {object} [opts]
+ * @param {Pattern} [opts.chainInfoValueShape] legacy support for contracts
+ *   deployed with `valueShape: CosmosChainInfoShape`
  */
-export const makeChainHub = (zone, agoricNames, vowTools) => {
+export const makeChainHub = (zone, agoricNames, vowTools, opts = {}) => {
   /** @type {MapStore<string, ChainInfo>} */
   const chainInfos = zone.mapStore('chainInfos', {
     keyShape: M.string(),
-    valueShape: ChainInfoShape,
+    valueShape: opts.chainInfoValueShape || ChainInfoShape,
```

Comment:
reasonable solution.

consider putting `ChainInfoShape` into the default args.

please document what happens if a contract fails to set this value.